wmfdata/hive.py

```diff
@@ -1,25 +1,45 @@
-import datetime as dt
 import os
-import re
-from shutil import copyfileobj
-import subprocess
+import pwd
 import tempfile
-import warnings
-
 import pandas as pd
 
+from pyhive import hive
+from shutil import copyfileobj
 from wmfdata.utils import (
-    check_kerberos_auth,
     ensure_list
 )
 
+HIVE_URL = 'analytics-hive.eqiad.wmnet'
+KERBEROS_SERVICE_NAME = 'hive'
+
+
 def run_cli(
-  commands, format = "pandas", heap_size = 1024, use_nice = True,
-  use_ionice = True
+    commands, format="pandas", heap_size=1024, use_nice=True,
+    use_ionice=True
 ):
     """
-    Runs SQL commands against the Hive tables in the Data Lake using Hive's
-    command line interface.
+    [DEPRECATED] Formerly ran SQL commands against the Hive tables in the Data Lake
+    using Hive's command line interface.  Currently just calls `run`.
+
+    Arguments:
+    * `commands`: the SQL to run. A string for a single command or a list of
+      strings for multiple commands within the same session (useful for things
+      like setting session variables). Passing more than one query is *not*
+      supported, and will usually result in an error.
+    * `format`: what format to return the results in
+        * "pandas": a Pandas data frame
+        * "raw": a TSV string, as returned by the command line interface.
+    * `heap_size`: the amount of memory available to the Hive client. Increase
+      this if a command experiences an out of memory error.
+    * `use_nice`: [Deprecated]
+    * `use_ionice`: [Deprecated]
+    """
+    return run(commands, format, heap_size)
+
+
+def run(commands, format="pandas", heap_size=1024):
```

Comment:
It looks like the `heap_size` argument isn't used. If it's not useful to implement that feature (I'm not sure), we should mark it as deprecated and log a warning.

We should also keep the `engine` parameter, with a deprecation warning.

Reply:
I couldn't figure out how to set the heap_size through pyhive, I think I'm missing something fundamental.  I asked for help and nobody else knew, but I don't think we should deprecate it.  I'll just add a warning that we're not setting it.

(done on engine)

Reply:
Hmm. I did some searching and I'm starting to think that setting the heap size isn't relevant when we're not using the `hive` CLI. That setting controls the memory used by a Hadoop process, so setting it on a worker in the Hadoop cluster always makes sense. But in this case, I think it was actually controlling the memory used by the _local_ Hadoop process started by the `hive` CLI, which was a thick client. But with `beeline` and PyHive, I don't think any processing actually happens on the client, which would mean there are no client-side memory settings to configure.

So that's my suspicion, which would mean it makes sense to deprecate the option. But I definitely could be misunderstanding the architecture.

Reply:
You can still control the memory used by the cluster, I think.  So I was thinking of finding a way to do that, but you're right that heap_size wouldn't be the right name for that anyway.  You're right, deprecating.